src/shogun/regression/LeastAngleRegressionImpl.cpp

```diff
@@ -40,23 +40,23 @@ using namespace Eigen;
 
 namespace shogun
 {
-SGMatrix<float64_t> CLeastAngleRegressionImpl::cholesky_insert_helper(SGMatrix<float64_t> X, SGMatrix<float64_t> R, VectorXd x_vec, SGVector<float64_t> active_set)
+SGMatrix<float64_t> CLeastAngleRegressionImpl::cholesky_insert_helper(Ref<MatrixXd> map_X, SGMatrix<float64_t>& R, int32_t i_max_corr, SGVector<float64_t> active_set)
```

Comment:
@Saurabh7 maybe this is the reason why in the initial implementation there was that transposition of the input matrix. but it'd be good to actually use a more optimal solution for matrix transposition than nested loops.
all lapack/blas functions has `TRANS` function arg that will let you do the operation on the original or the transposed version of the matrix. would be interesting to see the cache misses there ... :)